### PR TITLE
binfmt/loadable: move binary_s to stack to avoid access allocator

### PR DESCRIPTION
**Pull request by anchao**
## Summary

binfmt/loadable: move binary_s to stack to avoid access allocator

## Impact

binfmt/loadable: move binary_s to stack to avoid access allocator

Improve performance by reducing allocator accesses

Signed-off-by: chao an <anchao@lixiang.com>

## Testing

sim/nsh

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements. While it provides a summary and testing information, it lacks crucial details. Here's a breakdown of what's missing:

* **Summary:**  While the title mentions moving `binary_s` to the stack, it doesn't explain *why* this is necessary. Is there a specific problem being solved (e.g., performance bottleneck, memory leak, etc.)? What part of `binfmt/loadable` is affected? How exactly does moving `binary_s` to the stack work to address the issue?  Are there any related NuttX issues?

* **Impact:**  The current impact statement simply repeats the summary.  It needs to address *all* the listed points:
    * **Is new feature added? Is existing feature changed?**:  This is a change to an existing feature.
    * **Impact on user:** Likely NO, but should be explicitly stated.
    * **Impact on build:** Likely NO, but should be explicitly stated.
    * **Impact on hardware:** Likely NO, but should be explicitly stated.
    * **Impact on documentation:**  If the change affects how `binfmt/loadable` works internally, documentation might need updates.  This needs to be clarified.
    * **Impact on security:**  Needs to be addressed.  Does this change have any security implications?
    * **Impact on compatibility:** Needs to be addressed. Does this change break backward compatibility?
    * **Anything else to consider?:**  Anything specific that reviewers should be aware of?

* **Testing:** While the target (`sim/nsh`) is specified, the host build environment details are missing (OS, CPU, Compiler). More importantly, the "Testing logs before change" and "Testing logs after change" sections are empty.  Concrete evidence demonstrating the improvement (e.g., performance benchmarks, memory usage before and after) is essential.  Just stating "Improve performance" is not sufficient.  Quantify the improvement.


In short, the PR needs to be significantly more detailed and provide concrete evidence of the change's benefits and address all the points in the template.

File: binfmt/binfmt_exec.c [hunk 1]
```diff
@@ -80,24 +80,30 @@ static int exec_internal(FAR const char *filename,
                          FAR const posix_spawn_file_actions_t *actions,
                          FAR const posix_spawnattr_t *attr, bool spawn)
 {
-  FAR struct binary_s *bin;
+  FAR struct binary_s *pbin;
```
Comment (xiaoxiang781216) [inline binfmt/binfmt_exec.c:83]: let's keep name as before to reduce the modification

Comment (anchao) [reply binfmt/binfmt_exec.c:83]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by fdcavalcanti** [comment]
We are having some CI failure on `esp32s3-devkit:adb` and `esp32s3-devkit:txtable` after this PR. It builds ok but Nuttx crashes during startup.

@anchao any ideas?

**Comment by anchao** [comment]
> We are having some CI failure on `esp32s3-devkit:adb` and `esp32s3-devkit:txtable` after this PR. It builds ok but Nuttx crashes during startup.
> 
> @anchao any ideas? I'll take a look soon.

emm... Sorry for the break, maybe the nsh stack overflowed, let me check the configuration

**Comment by anchao** [comment]
@fdcavalcanti please help to verify this PR: https://github.com/apache/nuttx/pull/15160
